library/std/src/sys/pal/uefi/process.rs

```diff
@@ -449,20 +455,22 @@ mod uefi_command_internal {
             }
         }
 
-        pub fn set_args(&mut self, args: &OsStr) {
+        pub fn set_args(&mut self, mut args: Box<[u16]>) {
             let loaded_image: NonNull<loaded_image::Protocol> =
                 helpers::open_protocol(self.handle, loaded_image::PROTOCOL_GUID).unwrap();
 
-            let mut args = args.encode_wide().collect::<Vec<u16>>();
-            let args_size = (crate::mem::size_of::<u16>() * args.len()) as u32;
+            let len = args.len();
+            let ptr = args.as_mut_ptr();
+            crate::mem::forget(args);
```

Comment:
This invalidates `ptr` as it moves the `Box`. Use `Box::into_raw` instead:
```suggestion
            let ptr = Box::into_raw(args).as_mut_ptr();
```